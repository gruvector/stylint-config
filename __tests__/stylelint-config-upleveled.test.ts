import { $ } from 'execa';
import { expect, test } from 'vitest';

test('Errors correctly reported with Stylelint config', async () => {
  const { exitCode, stdout, stderr } = await $({
    reject: false,
  })`pnpm stylelint '**/*.{css,scss,less,js,tsx}'`;
  expect(exitCode).toBe(2);
  expect(stdout).toMatchSnapshot();
  expect(stderr).toMatchSnapshot();
});
