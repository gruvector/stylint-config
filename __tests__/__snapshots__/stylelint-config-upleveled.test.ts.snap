// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Errors correctly reported with Stylelint config 1`] = `""`;

exports[`Errors correctly reported with Stylelint config 2`] = `
"
__tests__/index.module.scss
  [2m1:1[22m   [31m[31mâœ–[39m  Unexpected unknown type selector "lol"         [2mselector-type-no-unknown[22m
  [2m1:5[22m   [31m[31mâœ–[39m  Unexpected empty block                         [2mblock-no-empty[22m
  [2m5:3[22m   [31m[31mâœ–[39m  Unexpected unknown property "backgound-color"  [2mproperty-no-unknown[22m
  [2m7:3[22m   [31m[31mâœ–[39m  Unexpected duplicate "--custom-property"       [2mdeclaration-block-no-duplicate-custom-properties[22m
 [2m16:9[22m   [31m[31mâœ–[39m  Unexpected unknown type selector "c"           [2mselector-type-no-unknown[22m
 [2m16:13[22m  [31m[31mâœ–[39m  Unexpected unknown type selector "d"           [2mselector-type-no-unknown[22m
 [2m16:15[22m  [31m[31mâœ–[39m  Unexpected unknown type selector "e"           [2mselector-type-no-unknown[22m

7 problems ([31m7 errors[39m, [33m0 warnings[39m)
"
`;
